src/main/java/com/hms_networks/americas/sc/extensions/system/application/SCAppManagement.java

```diff
@@ -56,4 +61,72 @@ public static void disableAppAutoRestart() {
     Logger.LOG_DEBUG("Disabling next JVM run command.");
     RuntimeControl.configureNextRunCommand(jvmCommand);
   }
+
+  /**
+   * Checks if a valid WAN IP address has been assigned to the device.
+   *
+   * @return true if an initialized WAN IP address is found.
+   */
+  public static boolean deviceHasWanIP() {
+    final String uninitializedWanIp = "0.0.0.0";
+    return NetManager.getWanIp().equalsIgnoreCase(uninitializedWanIp);
+  }
+
+  /**
+   * Waits for a number of minutes for a WAN IP to be establlshed if one does not already exist.
+   * After the given time has passed, returns false indicating no WAN IP is present.
+   *
+   * @param timeOutMinutes returns true if an initialized WAN IP is detected
+   */
+  public static boolean waitForWanIp(int timeOutMinutes) throws InterruptedException {
```

Comment:
What are your thoughts on making this seconds instead of minutes? 

I feel like down the road we might end up wanting/needing to tweak this value. If we establish that waitForWanIp(*int*) is minutes-based, it will be challenging to switch to seconds later to allow for better precision, as that would break any existing usages/calls to this method.

Reply:
For example, this would make the method super flexible and allow any time unit: 

```
  /**
   * Waits for a number of minutes for a WAN IP to be established if one does not already exist.
   * After the given time has passed, returns false indicating no WAN IP is present.
   *
   * @param timeout the timeout value in the specified time unit
   * @param timeUnit the time unit of the timeout value
   * @return true if an initialized WAN IP is detected, false otherwise
   */
  public static boolean waitForWanIp(int timeout, SCTimeUnit timeUnit) throws InterruptedException {
    // Convert timeout to seconds
    long timeoutSeconds = timeUnit.toSeconds(timeout);

    // Get rounded up number of wait periods in timeout (double cast to avoid integer division)
    int maxWaitPeriods = (int) Math.ceil((double) timeoutSeconds / (double) secondsInWaitPeriod);
```

Reply:
I like this solution a lot better. The time unit flexibility is much more ideal as an API method